AIX build broken with OpenXL ( ibm-clang )

```
/opt/IBM/openxlC/17.1.2/bin/ibm-clang_r -c -O2 -DSMALL_MATRIX_OPT -DGEMM_GEMV_FORWARD -DSBGEMM_GEMV_FORWARD -DMAX_STACK_ALLOC=2048 -fopenmp -DHAVE_P10_SUPPORT -Wall -DF_INTERFACE_IBM -fPIC -DDYNAMIC_ARCH -DSMP_SERVER -DUSE_OPENMP -DNO_WARMUP -DMAX_CPU_NUMBER=16 -DMAX_PARALLEL_NUMBER=1 -DBUILD_BFLOAT16 -DBUILD_SINGLE=1 -DBUILD_DOUBLE=1 -DBUILD_COMPLEX=1 -DBUILD_COMPLEX16=1 -DVERSION=\"0.3.30.dev\" -O3 -mcpu=power10 -mtune=power10 -mvsx -fno-fast-math -DUSE_OPENMP -fopenmp -fno-integrated-as -m64 -UASMNAME -UASMFNAME -UNAME -UCNAME -UCHAR_NAME -UCHAR_CNAME -DASMNAME=dgemv_n_POWER10 -DASMFNAME=dgemv_n_POWER10_ -DNAME=dgemv_n_POWER10_ -DCNAME=dgemv_n_POWER10 -DCHAR_NAME=\"dgemv_n_POWER10_\" -DCHAR_CNAME=\"dgemv_n_POWER10\" -DNO_AFFINITY -DTS=_POWER10 -I.. -DBUILD_KERNEL -DTABLE_NAME=gotoblas_POWER10 -DDOUBLE  -UCOMPLEX -DDOUBLE  -UCOMPLEX -DDOUBLE -UCOMPLEX  -UTRANS ../kernel/power/dgemv_n_power10.c -o dgemv_n_POWER10.o

In file included from ../kernel/power/dgemv_n_power10.c:30:
../kernel/power/dgemv_n_microk_power10.c:469:13: error: dereference of pointer to incomplete type 'double[]'
  469 |        "m" (*(double (*)[]) ap),
      |             ^
1 error generated.
make[1]: *** [Makefile.L2:312: dgemv_n_POWER10.o] Error 1
make[1]: *** Waiting for unfinished jobs....
1 warning generated.
```

--- a/kernel/power/dgemv_n_microk_power10.c
+++ b/kernel/power/dgemv_n_microk_power10.c
@@ -466,7 +466,7 @@ static void dgemv_kernel_4x8 (long n, double *ap, long lda, double *x, double *y
        "=b" (tmp)
      :
        "m" (*(double (*)[4]) x),
-       "m" (*(double (*)[]) ap),
+       "m" (*(double (*)[4]) ap),
        "d" (alpha),	// 14
        "r" (x),		// 15
        "3" (ap),	// 16

